feat: autoSelectDefault and showUsageStatus props of StorageSelector

ironAiken2 opened the pull request:
<!--
Please precisely, concisely, and concretely describe what this PR changes, the rationale behind codes,
and how it affects the users and other developers.
-->
### This PR resolves #2151 

feature:
- autoSelectDefault
:  set default selected host, "default" | "usage" or no use. 
- showUsageSelect
: "true" for showing usage in each host, "false" for disabling the feature.
- useControllableValue
: StorageSelect component can be controllable or not. If the parent passes state, it uses it; otherwise, it uses its own state.

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

ironAiken2 commented:
But I thing that the upgrading version issue covers a large area and should be fixed by addressing the issue in the current version of the project.

yomybaby commented on react/src/components/StorageSelect.tsx:
```diff
@@ -0,0 +1,118 @@
+        }}
+        options={options}
+        optionRender={(option) => (
+          <Space>
```
Please use 'Flex' instead of `<Space/>` if available.

yomybaby commented on react/src/components/StorageSelect.tsx:
```diff
@@ -0,0 +1,118 @@
+          <Space>
+            {option.label}
+            {showUsageStatus && (
+              <div
```
How about '[Badge](https://ant.design/components/badge#badge-demo-status)'?

yomybaby commented on react/src/components/StorageSelect.tsx:
```diff
@@ -0,0 +1,118 @@
+  showUsageStatus?: boolean;
+}
+const StorageSelect: React.FC<Props> = ({
+  value,
```
Let's use [ahooks's useControllerbleValue](https://ahooks.js.org/hooks/use-controllable-value)

yomybaby commented on react/src/components/StorageSelect.tsx:
```diff
@@ -0,0 +1,118 @@
+  return (
+    <>
+      <style>{rawSelectCss}</style>
+      <Select
```
Form controls such as Select, Input, etc should return one element for `Form.Item`.

lizable commented on react/src/components/StorageSelect.tsx:
```diff
@@ -0,0 +1,117 @@
+          <Flex justify="between" align="center">
+            <Badge
+              // @ts-ignore
+              status={status[idx]}
```
We already have color setting function in index. Please refer to this code below and update Badge component props.
https://github.com/lablup/backend.ai-webui/blob/ff29a586a1e7af517289fa1999450692f638282e/react/src/helper/index.tsx#L227-L233

lizable commented on react/src/components/StorageSelect.tsx:
```diff
@@ -0,0 +1,117 @@
+  }, [vhostInfo]);
+
+  const optionRender = useMemo(() => {
+    const status = ['success', 'warning', 'error'];
```
And I think it's not necessary if we use color instead.

lizable reviewed (approved):
I just found some minor test failures in github action, originated from dependency array in hooks([useEffect](https://github.com/lablup/backend.ai-webui/pull/2180/files/a8537491593776c31e4485e01c4163b2d5fd87e5#diff-bec9f3b247068290d015fe25aac131df101c7055e4a8d8276085abd1f90ed04cR86) and [useMemo](https://github.com/lablup/backend.ai-webui/pull/2180/files/a8537491593776c31e4485e01c4163b2d5fd87e5#diff-bec9f3b247068290d015fe25aac131df101c7055e4a8d8276085abd1f90ed04cR66)). You could either follow the guidance from github action or just input right above the line if there has certain reason (https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528).

lizable reviewed (changes requested):
Please resolve test failure on github action following the comments above.

ironAiken2 commented:
I thought the "vhostInfo" data is used directly within the useEffect, and changes in it have a significant impact on determining the selectd host. So I added vhostInfo to the dependencies array. 

However, the other items are only configurable in code, so I removed them from the Dependencies array because I thought they were less likely to be changed because the user can't change their settings.

lizable reviewed (approved):
LGTM.

yomybaby commented on react/src/components/StorageSelect.tsx:
```diff
@@ -0,0 +1,120 @@
+      options={_.map(vhostInfo?.allowed, (host) => ({
+        label: showUsageStatus ? (
+          <Flex align="center" gap={'xs'}>
+            {/* TODO: add tooltip for '여유/주의/부족' */}
+            {vhostInfo?.volume_info[host]?.usage && (
+              <Tooltip
+                title="여유 or 주의 or 부족"
+                // @ts-ignore
+                getPopupContainer={() => shadowRoot}
```
I push a commit to add a tooltip with `getPopupContainer`. Please add a proper text for each case. @ironAiken2

yomybaby commented:
I pushed commits to display search highlights and update type definition. @ironAiken2 , Please check.
I merged this PR. Thank you!
